modules/auth/auth.controller.js

```diff
@@ -0,0 +1,296 @@
+    return res.status(200).json({success: 1, message: globalConstants.responseMessages.signUpUser.success, data: req.user});
+  } else {
+    logger.error(authConstants.labels.signupCtrl.failure, req);
+    return res.status(200).json({success: 0, message: globalConstants.responseMessages.signUpUser.failure, data: null});
+  }
+};
+
+/**
+ * Created logout controller that removs the user session
+ * @param {Object} req
+ * @param {Object} res
+*/
+const logoutCtrl = (req, res) => {
+  req.logOut(function(err) {
+    if (err) {
+      logger.error(authConstants.labels.logoutCtrl.failure, req, err);
+      return res.status(406).json({success: 0, message: authConstants.responseMessages.logOutUser.failure, data: null});
+    }
+    logger.info(authConstants.labels.logoutCtrl.success);
+    return res.status(200).json({success: 1, message: authConstants.responseMessages.logOutUser.success, data: null});
+  });
+};
+
+/**
+ * Created controller to change password
+ * @param {Object} req
+ * @param {Object} res
+ * @return {Object}
+ */
+const changePasswordCtrl = async (req, res) => {
+  if (!req.user) {
+    logger.error(authConstants.labels.changePasswordCtrl.noUserInReq, req.user);
+    return res.status(406).json({success: 0, message: globalConstants.responseMessages.loggedInUser.failure, data: null});
+  }
+
+  const password = req.body.password;
+  const newPassword = req.body.newPassword;
+  const [err, user] = await to(User.findById({_id: req.user._id}));
+
+  if (err) {
+    logger.error(authConstants.labels.changePasswordCtrl.findingUser.failure, err);
+    return res.status(406).json({success: 0, message: authConstants.responseMessages.changePassword.failure, data: null});
+  }
+
+  if (!user) {
+    return res.status(406).json({success: 0, message: authConstants.responseMessages.changePassword.wrongCredentails, data: null});
+  }
+
+  if (user.password) {
+    if (!(await user.isPasswordMatched(password))) {
+      return res.status(406).json({success: 0, message: authConstants.responseMessages.changePassword.wrongCredentails, data: null});
+    }
+  } else {
+    return res.status(409).json({success: 0, message: authConstants.responseMessages.changePassword.conflict, data: null});
+  }
+
+  logger.info(authConstants.labels.changePasswordCtrl.savingNewPassword.savingMsg, req.user);
+  user.password = newPassword;
+  const [passwordSavingErr, passwordSaved] = await to(user.save());
+
+  if (passwordSavingErr) {
+    logger.error(authConstants.labels.changePasswordCtrl.savingNewPassword.failure, passwordSavingErr);
+    return res.status(501).json({success: 0, message: authConstants.responseMessages.changePassword.failure, data: null});
+  }
+
+  if (passwordSaved) {
+    logger.info(authConstants.labels.changePasswordCtrl.savingNewPassword.success, req.user);
+    return res.status(200).json({success: 1, message: authConstants.responseMessages.changePassword.success, data: user});
+  } else {
```

Comment:
```suggestion
```
No need to write `else` as you already added the return statement in `if`

Reply:
done